xwiki-rendering-macros/xwiki-rendering-macro-toc/src/main/java/org/xwiki/rendering/internal/macro/toc/TocMacro.java

```diff
@@ -123,7 +130,15 @@ public boolean supportsInlineMode()
 
         TreeParametersBuilder builder = new TreeParametersBuilder();
         TreeParameters treeParameters = builder.build(rootBlock, parameters, context);
-        return this.tocTreeBuilder.build(treeParameters);
+        List<Block> parametersList = this.tocTreeBuilder.build(treeParameters);
```

Comment:
The variable naming is a bit strange since this generates the list of Block for the output, they're not parameters. Proposed variable name: `macroBlocks` or `result`.